app/lib/actions/candlepin/consumer/unattach_subscriptions.rb

```diff
@@ -0,0 +1,31 @@
+module Actions
+  module Candlepin
+    module Consumer
+      class UnattachSubscriptions < Candlepin::Abstract
+        input_format do
+          param :uuid, String
+          param :entitlements, Array
+        end
+
+        def plan(system, subscriptions)
+          entitlements = []
+          subscriptions.each do |subscription|
+            system.entitlements.each do |entitlement|
+              if subscription[:id] == entitlement[:pool][:id]
+                entitlements << entitlement[:id]
+                break
+              end
+            end
+          end
```

Comment:
What happens if you try to remove an entitlement that is not attached to the content host at present? Does it throw an error?